pkg/node/manager.go

```diff
@@ -72,7 +72,8 @@ func UpdateNode(ni Identity, n *Node) {
 
 	nodeIP := n.GetNodeIP(false)
 	updateNodeCIDR(n.IPv4AllocCIDR, nodeIP)
-	updateNodeCIDR(n.IPv6AllocCIDR, nodeIP)
+	nodeIP6 := n.GetNodeIP(true)
+	updateNodeCIDR(n.IPv6AllocCIDR, nodeIP6)
```

Comment:
I'm pretty sure that this will break the datapath because we do:
```
        node_id = bpf_htonl(tunnel->ip4);
        key.tunnel_id = seclabel;
        key.remote_ipv4 = node_id;

        cilium_trace(skb, DBG_ENCAP, node_id, seclabel);

        ret = skb_set_tunnel_key(skb, &key, sizeof(key), 0);
```

It's perfectly fine to encapsulate IPv6 over IPv4 here.